project_timesheet_management/models/timesheet_invoice.py

```diff
@@ -0,0 +1,764 @@
+            ):
+                total_amount = sum(
+                    (line.worked_hours * line.resource_amount)
+                    for line in timesheet_invoice.resource_calculation_ids
+                )
+                timesheet_invoice._write({"total_amount": total_amount})
+            if self._context.get("from_approval", False):
+                next(iter(timesheet_invoice.message_ids)).write(
+                    {
+                        "author_id": user_obj.sudo()
+                            .browse(request.session.uid)
+                            .partner_id.id
+                    }
+                )
+        return res
+
+    @api.model_create_multi
+    def create(self, vals):
+        """
+        @override
+        To calculate the amount when creating the record.
+        """
+        res = super(TimesheetInvoice, self).create(vals)
+        po_group = 'project_management_security.group_project_coordinator'
+        mail_template = self.env.ref(
+            'project_timesheet_management.email_timesheet_invoice_confirmed')
+        for timesheet_invoice in res:
+            if timesheet_invoice.document_type == "timesheet":
+                total_amount = sum(
+                    (line.worked_hours * line.resource_amount)
+                    for line in timesheet_invoice.resource_calculation_ids
+                )
+                res.write({"total_amount": total_amount})
+                user_ids = timesheet_invoice.project_id.favorite_user_ids.filtered(
+                    lambda user: user.has_group(po_group)) or \
+                    timesheet_invoice.project_id.resource_ids.mapped(
+                        'user_ids').filtered(lambda user: user.has_group(po_group))
+                mail_state = 'Uploaded'
+                for user_id in user_ids:
+                    mail_template.write({
+                        'email_to': user_id.email or '',
+                        'email_from': self.env.user.email or '',
+                        'reply_to': user_id.email or ''})
+                    mail_template.sudo().with_context(state=mail_state).send_mail(
+                        timesheet_invoice.id, force_send=True)
+        return res
+
+    def button_rejected(self):
+        """
+        Set timesheet invoice to Reject State.
+        Updating with sudo env due to the access rights.
+        """
+        mail_template = self.env.ref(
+            "project_timesheet_management.email_timesheet_invoice_confirmed"
+        )
+
+        po_group = 'project_management_security.group_supp_project_coordinator'
+        cann_po_group = 'project_management_security.group_project_coordinator'
+        # )
+        author_id = (
+            self.env["res.users"].sudo().browse(request.session.uid).partner_id.id
+        )
+        for timesheet_inv in self:
+            u_lst = []
+            mail_status = 'Rejected'
+            if timesheet_inv.state == "pre-approved":
+                mail_template.write({
```

Comment:
You should not write to the email template, instead you can use email_values as parameter for send_mail function and no need to define reply_to  separately as it uses the same value as email_from